src/test/java/com/direwolf20/buildinggadgets/building/placementTests/SingleTypeProviderTest.java

```diff
@@ -0,0 +1,51 @@
+package com.direwolf20.buildinggadgets.building.placementTests;
+
+import com.direwolf20.buildinggadgets.common.building.placement.SingleTypeProvider;
+import com.direwolf20.buildinggadgets.util.UniqueBlockState;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import org.junit.jupiter.api.*;
+
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SingleTypeProviderTest {
+
+    private final IBlockState state = UniqueBlockState.createNew();
+    private final SingleTypeProvider provider = new SingleTypeProvider(state);
+    private final Random random = new Random();
+
+    //TODO implement custom runner that launches minecraft
```

Comment:
Again, needs to be done before merge.

Reply:
That might be too much in my opinion... 
I'm already working on abstracting things enough in the API, so that this is not necessary, which is in my opinion the better solution, than to write something that makes the Tests depending on GameStart just because the API-Design is not capable of running on it's own. In theory a good API can be tested without depending on the environment it is used in. Going that far will propably make this a larger PR than good for it...

Reply:
Or is writing a dedicated MCTestRunner so easy that it can go below a few hundred lines...?

Reply:
The api part can run on its own if Mojang isn't that stupid to integrate every little function, including sound in the block object...
Or we can just get rid of JUnit completely like what Forge does

Reply:
Junit makes for very convenient Tests, in contrast to what Forge does...
I'd rather keep it...

Reply:
Since Forge already provides `ASMData` on preInit, we can create a fake JUnit environment, where we have `@Test` to indicate a method need to be ran and `@RepeatedTest(value = int)` to indicate a method need to be ran multiple times.
This should be very simple (but should not be in this PR, though) and would work pretty well, in my opinion.